ndcube/ndcollection.py

```diff
@@ -65,12 +65,19 @@ def __init__(self, key_data_pairs, aligned_axes=None, meta=None, **kwargs):
                 f"__init__() got an unexpected keyword argument: '{list(kwargs.keys())[0]}'"
             )
         # Attach aligned axes to object.
-        self.aligned_axes = aligned_axes
-        if self.aligned_axes is None:
+        self._aligned_axes = aligned_axes
+        if self._aligned_axes is None:
             self.n_aligned_axes = 0
         else:
             self.n_aligned_axes = len(self.aligned_axes[keys[0]])
 
+    @property
+    def aligned_axes(self):
+        """
+        The axes of each array that are aligned in numpy order.
+        """
+        return self._aligned_axes
```

Comment:
So I for the life of me could not get this attribute to render in sphinx without doing this.

Now, this has to be wrong, there has to be a way to get the `self.whatever` to render in the docs and I don't know what that is.

Reply:
I wonder if changing this to a property should be classes as an API change.  The tests seem to be passing do presumably this doesn't cause internal errors.  But if someone is altering `self.aligned_axes in place, this will cause their workflow to break.  That said, we probably don't want anyone changing those values in place.

Perhaps a breaking changelog entry should be added to this PR saying `self.aligned_axes` is now a property.

Reply:
I can add a setter and that should avoid a breaking change?